docs(gatsby-link): Added some information regarding the limitations with the link package (#11909)
gatsbyjs/gatsby: docs/docs/gatsby-link.md

```

[egghead]: https://egghead.io/playlists/use-gatsby-s-link-component-to-improve-site-performance-and-simplify-site-development-7ed3ddfe

> ## Limitations when appending query parameters or hash links programmatically

Reviewer: General question - what about query parameters makes them untenable with the `Link` component?

This seems like a perfectly reasonable use-case, e.g.

```jsx
<Link to="/user?id=1234">Go to 1234 user</Link>
```

Is there more nuance here? Is it query param navigation on the _same_ page that doesn't work?

Author: Yes, it wouldn't work if you did `<Link to="?id=1234">Go to 1234 user</Link>`. For example, if I built a Tab control that used the query parameter for the current tab.

Reviewer: Let's maybe dial back the language a bit then? 

Maybe something like:

```suggestion
## Recommendations for programmatic, in-app navigation
```

> Neither `<Link>` not `navigate` can be used for in-route navigation with a hash or query parameter. If you need this behavior, you should either use an anchor tag or import the `@reach/router` package to make use of its `navigate` function, like so:

Reviewer: ```suggestion
Neither `<Link>` nor `navigate` can be used for in-route navigation with a hash or query parameter. If you need this behavior, you should either use an anchor tag or import the `@reach/router` package--which Gatsby already depends upon--to make use of its `navigate` function, like so:
```

> Neither `<Link>` not `navigate` can be used to add query parameters or send your user to a hash link due to their use of `withPrefix`. If you need this behavior, you should either use an anchor tag or load the `@reach/router` package to make use of its `navigate` function.

Reviewer: ```suggestion
Neither `<Link>` not `navigate` can be used for in-route navigation with a hash or query parameter. If you need this behavior, you should either use an anchor tag or import the `@reach/router` package to make use of its `navigate` function, like so:
```
